File: mobile/library/kotlin/src/io/envoyproxy/envoymobile/filters/FilterTrailerStatus.kt
package io.envoyproxy.envoymobile

/*
 * Status returned by filters when transmitting or receiving trailers.
 */
// TODO: create abstract Trailers class.
sealed class FilterTrailerStatus<T : Headers> {
  /**
   * Continue filter chain iteration, passing the provided trailers through.
   */
  class Continue<T : Headers>(val trailers: T) : FilterTrailerStatus<T>()

  /**
   * Do not iterate to any of the remaining filters in the chain with trailers.
   *
   * Calling `continueRequest()`/`continueResponse()` MUST occur when continued filter iteration
   * is desired.
   */
  class StopIteration<T : Headers>(val trailers: T) : FilterTrailerStatus<T>()
}

File: mobile/library/swift/src/filters/FilterTrailerStatus.swift
/// Status returned by filters when transmitting or receiving trailers.
@frozen
public enum FilterTrailerStatus<T: Headers>: Equatable {
  /// Continue filter chain iteration, passing the provided trailers through.
  case `continue`(T)

  /// Do not iterate to any of the remaining filters in the chain with trailers.
  ///
  /// Calling `continueRequest()`/`continueResponse()` MUST occur when continued filter iteration
  /// is desired.
  case stopIteration(T)
}
